services/management/backup/backups_service.go

```diff
@@ -0,0 +1,161 @@
+// pmm-managed
+// Copyright (C) 2017 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+package backup
+
+import (
+	"context"
+
+	"github.com/AlekSi/pointer"
+	"github.com/pkg/errors"
+
+	backupv1beta1 "github.com/percona/pmm/api/managementpb/backup"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	"gopkg.in/reform.v1"
+
+	"github.com/percona/pmm-managed/models"
+)
+
+// BackupsService represents backups API.
+type BackupsService struct {
+	db          *reform.DB
+	jobsService jobsService
+}
+
+// NewBackupsService creates new backups API service.
+func NewBackupsService(db *reform.DB, jobsService jobsService) *BackupsService {
+	return &BackupsService{
+		db:          db,
+		jobsService: jobsService,
+	}
+}
+
+// StartBackup starts on-demand backup.
+func (s *BackupsService) StartBackup(ctx context.Context, req *backupv1beta1.StartBackupRequest) (*backupv1beta1.StartBackupResponse, error) {
+	var err error
```

Comment:
this err is used only in `func(tx *reform.TX) error`, so you could move it there

Reply:
It is used also below
`err = s.jobsService.StartMySQLBackupJob(job.ID, job.PMMAgentID, 0, req.Name, config, locationConfig)`